core/DCache.cpp

```diff
@@ -75,15 +78,19 @@ namespace olympia {
             if(!busy_) {
                 busy_ = true;
                 cache_pending_inst_ = memory_access_info_ptr;
-                uev_lookup_inst_.schedule(sparta::Clock::Cycle(cache_latency_)); // Replace with call to BIU
+                out_biu_req_.send(cache_pending_inst_->getInstPtr());
             }
         }
         out_lsu_lookup_ack_.send(memory_access_info_ptr);
     }
 
-    void DCache::lookupInst_() {
+    void DCache::getAckFromBIU_(const InstPtr &inst_ptr) {
+        out_lsu_lookup_req_.send(cache_pending_inst_);
+        reloadCache_(inst_ptr->getRAdr());
```

Comment:
We are reloading the cache twice.
Reload it once before sending the event 

@vineethkm

Reply:
I missed that. I'll make the changes